examples/inspection/plot_causal_interpretation.py

```diff
@@ -0,0 +1,150 @@
+"""
+=============================================================
+Failure of Machine Learning Models to infer causal effects
+=============================================================
+
+Machine Learning models are great for measuring statistical associations.
+Unfortunately, unless we're willing to make strong assumptions about the data,
+those models are unable to infer causal effects.
+
+To illustrate this, we will simulate a situation in which we try to answer
+one of the most important questions in economics of education:
+what is the causal effect of earning a college degree on hourly wages?
+Although the answer to this question is crucial to policy makers,
+`Omitted-Variable Biases <https://en.wikipedia.org/wiki/Omitted-variable_bias>`_ (OVB)
+prevent us from identifying that causal effect.
+
+
+.. contents::
+   :local:
+   :depth: 1
+"""
+
+print(__doc__)
+
+# %%
+# The dataset: simulated hourly wages
+# -----------------------------------
+#
+# The data generating process is laid out in the code below.
+# Work experience in years and a measure of ability are drawn
+# from a Normal distribution;
+# the hourly wage of one of the parents is drawn from Beta distribution.
+# We then create an indicator of college degree which is positivily
+# impacted by ability and parental hourly wage. Finally, we model
+# hourly wages as a linear function of all the previous variables and a
+# random component. Note that all variables have a positivie effect on
+# hourly wages.
+
+import numpy as np
+
+n_samples = 10000
+experiences = np.random.normal(20, 10, size=n_samples).astype(int)
+experiences[experiences < 0] = 0
+abilities = np.random.normal(0, 0.15, size=n_samples)
+parent_hourly_wages = 50 * np.random.beta(2, 8, size=n_samples)
```

Comment:
Yes good point. I guess I could set a np.random.seed at the beginning of the code. Is there any reason you would want to use randomState instead?

Reply:
As a programming style, I tend to avoid changing global state when possible. Also, most of the examples use some form of `np.random.RandomState(...)`.